src/main/java/com/google/alpollo/AnalysisInfoServlet.java

```diff
@@ -43,6 +43,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     try {
       analysisInfo = gson.fromJson(request.getReader(), AnalysisInfo.class);
       SongDataBase.saveAnalysisInfo(analysisInfo);
+      SongDataBase.saveSongRequest(analysisInfo.getSong());
     } catch (JsonSyntaxException | JsonIOException | IOException e) {
       response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage());
```

Comment:
You're catching the exceptions here to respond with an error, which is the correct thing to do. You can remove the `throws` declaration from the method signature. Same for `doGet`

Reply:
Redirecting it to @BudiGabe for #43

Reply:
Hmm VSCode told me I should throw the IOException too, maybe it's wrong this time.